vpc/service/long_lived_lock.go

```diff
@@ -19,6 +24,81 @@ const (
 	timeBetweenTryingToAcquireLocks = 15 * time.Second
 )
 
+func (vpcService *vpcService) readLocks(rows *sql.Rows) ([]*vpcapi.Lock, error) {
+	locks := []*vpcapi.Lock{}
+	for rows.Next() {
+
+		lock := vpcapi.Lock{}
+		var heldUntil time.Time
+		err := rows.Scan(&lock.Id, &lock.LockName, &lock.HeldBy, &heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		lock.HeldUntil, err = ptypes.TimestampProto(heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		locks = append(locks, &lock)
+	}
+
+	return locks, nil
+}
+
+func (vpcService *vpcService) GetLocks(ctx context.Context, empty *empty.Empty) (*vpcapi.GetLocksResponse, error) {
```

Comment:
Please start / stop a tracing span here, and use tracehelpers to set an error code on the span appropriately.